addons/purchase/models/purchase.py

```diff
@@ -525,6 +533,14 @@ def _compute_amount(self):
                 'price_subtotal': taxes['total_excluded'],
             })
 
+    @api.multi
+    def _compute_tax_id(self):
+        for line in self:
+            fpos = line.order_id.fiscal_position_id or line.order_id.partner_id.property_account_position_id
```

Comment:
I know that, and I don't say this is for UI operations, but for automatic operations that would use it for avoiding to make this manually (you are even calling this with a name that is candidate to be called from extension modules). This only prevents an incorrect setting.

The same logic is used with the line with `.filtered(lambda r: not line.company_id or r.company_id == line.company_id)`, so we should have both or none.